airo-camera-toolkit/airo_camera_toolkit/calibration/collect_calibration_data.py

```diff
@@ -0,0 +1,158 @@
+
+    Returns:
+        Path to the "data" subfolder of the calibration_dir.
+    """
+    if calibration_dir is None:
+        datetime_str = datetime.datetime.now().strftime("%Y-%m-%d_%H:%M:%S")
+        calibration_dir = os.path.join(os.getcwd(), f"calibration_{datetime_str}")
+
+    os.makedirs(calibration_dir, exist_ok=True)
+
+    data_dir = os.path.join(calibration_dir, "data")
+
+    # If data_dir already exists, check whether it is empty
+    if os.path.exists(data_dir) and len(os.listdir(data_dir)) != 0:
+        raise ValueError(f"The data subfolder of {calibration_dir} already exists and is not empty.")
+
+    os.makedirs(data_dir, exist_ok=True)
+    return data_dir
+
+
+def save_calibration_sample(
+    sample_index: int, robot: URrtde, camera: RGBCamera, data_dir: str
+) -> Tuple[HomogeneousMatrixType, OpenCVIntImageType]:
+    """Collect a single calibration sample and save to the data_dir.
+    A calibration data sample consists of an image and a TCP pose.
+
+    Args:
+        sample_index: index of the sample used for the names of the saved files.
+        robot: The robot being used to collect the data.
+        camera: The camera being used to collect the data.
+        data_dir: The directory to save the sample to.
+
+    Returns:
+        If charuco board detection is succesful, the TCP pose and the image, else None.
+    """
+    # Stop freedrive so robot is completely still at moment of the image capture
+    robot.rtde_control.endTeachMode()
+
+    ROBOT_STOP_WAIT_TIME = 0.5
+    time.sleep(ROBOT_STOP_WAIT_TIME)
+
+    image_rgb = camera.get_rgb_image_as_int()
+    image_bgr = ImageConverter.from_numpy_int_format(image_rgb).image_in_opencv_format
+
+    tcp_pose = robot.get_tcp_pose()
+
+    suffix = f"{sample_index:04d}"
+    image_filename = f"image_{suffix}.png"
+    tcp_pose_filename = f"tcp_pose_{suffix}.json"
+    image_filepath = os.path.join(data_dir, image_filename)
+    tcp_pose_filepath = os.path.join(data_dir, tcp_pose_filename)
+
+    cv2.imwrite(image_filepath, image_bgr)
+
+    pose = Pose.from_homogeneous_matrix(tcp_pose)
+    with open(tcp_pose_filepath, "w") as f:
+        json.dump(pose.model_dump(), f, indent=4)
+
+    robot.rtde_control.teachMode()
+
+    return tcp_pose, image_bgr
+
+
+def collect_calibration_data(robot: PositionManipulator, camera: RGBCamera, calibration_dir: Optional[str]) -> None:
+    """Collect calibration data samples for hand-eye calibration.
+
+    Args:
+        robot: the robot to use for collecting the data.
+        camera: the camera to use for collecting the data.
+        calibration_dir: directory to save the calibration data to, if None a directory will be created
+    """
+    from loguru import logger
+
+    data_dir = create_calibration_data_dir(calibration_dir)
+
+    logger.info(f"Saving calibration data to {data_dir}")
+    logger.info("Press S to save a sample, Q to quit.")
+
+    resolution = camera.resolution  # type: ignore
```

Comment:
why this type.ignore?

Reply:
Forgot to remove this one (was because ZED and Realsense had different types for this).